### PR TITLE
refactor router initialization

### PR DESCRIPTION
**Pull request by KATT**
- refactor router initialization in order to simplify work with #545

(sorry for all the force pushes to the branch, was fighting with CI)

### CODE REVIEW

File: packages/server/src/router.ts [hunk 1]
```diff
@@ -138,24 +144,18 @@ export class Router<
   }>;
```
Comment (KATT) [inline packages/server/src/router.ts:144]: Add `transformer:` up here

File: packages/server/src/router.ts [hunk 2]
```diff
@@ -138,24 +144,18 @@ export class Router<
...
-        defaultShape: DefaultErrorShape;
-      }) => {
-        return defaultShape;
-      }) as any,
+    this._def = {
+      queries: (def?.queries ?? safeObject()) as TQueries,
+      mutations: (def?.mutations ?? safeObject()) as TMutations,
+      subscriptions: (def?.subscriptions ?? safeObject()) as TSubscriptions,
+      middlewares: def?.middlewares ?? [],
+      errorFormatter: def?.errorFormatter ?? (defaultErrorFormatter as any),
```
Comment (KATT) [inline packages/server/src/router.ts:158]: and here

File: packages/server/src/router.ts [hunk 3]
```diff
@@ -490,16 +487,15 @@ export class Router<
...
       TContext,
       TQueries,
       TMutations,
       TSubscriptions,
       ReturnType<TErrorFormatter>
-    >({
-      ...this._def,
-      errorFormatter,
-    });
+    >;
```
Comment (KATT) [inline packages/server/src/router.ts:498]: and a `transformer()` function similar to this

**Review by KATT — COMMENTED** [review]
the only tricky part with data transformers is how to deal with output emitted from subscriptions

cc @delaneyj

**Review by delaneyj — APPROVED** [review]

**Pull request merged**